SharedSequence/src/main/kotlin/org/notests/sharedsequence/DriverTraits.kt

```diff
@@ -21,7 +22,7 @@ object DriverTraits : SharingTrait {
 }
 
 fun <Element> Driver<Element>.drive(onNext: (Element) -> Unit = {}): Disposable =
-  this.asObservable().subscribe(onNext, {})
+  this.asObservable().subscribe(onNext) { ErrorReporting.report(it) }
```

Comment:
@kzaher is this what you would expect? 

On iOS the onNext method in a `driver` cannot throw, but in Kotlin can. So when the `onNext` throws, `onError` will be called. When that happens what is the best scenario:
 1. we crash the app 
 2. we report an error in the error reporting tool (like in this PR)
 3. 1 + we add another `drive` method that accepts both `onNext` and `onError`.

Reply:
I prefer option 2. Or option 1. But I'm against 3 and 4 because what's the point then :D But that's just my opinion 🤷‍♂️

Reply:
I changed this. This approach is actually not good because it results in unsubscribing from the upstream. Check the last commit I force pushed.